### PR TITLE
Dispensing Log Report 

### PR DESCRIPTION
**Pull request by dbaluku**
https://app.asana.com/0/1133167201254915/1166636667807305/f

### CODE REVIEW

File: omod/src/main/resources/apps/ugandaemr_reports_app.json [hunk 1]
```diff
@@ -844,6 +844,14 @@
         "url": "reportingui/runReport.page?reportDefinition=662d4c00-d6bb-4494-8180-48776f415802",
         "order":10,
         "requiredPrivilege": "App: ugandaemrreports.recencyHTSClientCardDataExport"
+      },{
+        "id": "Dispensing Log Report ",
+        "extensionPointId": "org.openmrs.module.ugandaemr.integrationdataexports2019",
+        "type": "link",
+        "label": "Dispensing Log Report",
```
Comment (solemabrothers) [inline omod/src/main/resources/apps/ugandaemr_reports_app.json]: @dbaluku simply call it Dispensing Log

File: omod/src/main/resources/apps/ugandaemr_reports_app.json [hunk 2]
```diff
@@ -844,6 +844,14 @@
...
+        "url": "reportingui/runReport.page?reportDefinition=22a58e5e-2535-44f3-a710-3856154ca29f",
+        "order":15,
+        "requiredPrivilege": "App: ugandaemrreports.recencyHTSClientCardDataExport"
```
Comment (solemabrothers) [inline omod/src/main/resources/apps/ugandaemr_reports_app.json]: @dbaluku Why did you attach this role in perticular

File: api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/DispensingDatasetDefinitionEvaluator.java [hunk 3]
```diff
@@ -63,70 +65,71 @@ public DataSet evaluate(DataSetDefinition dataSetDefinition, EvaluationContext c
...
-                    }
-                    pdh.addCol(row, "date", DateUtil.formatDate(DateUtil.parseYmd(String.valueOf(o[5])), "yyyy-MM-dd"));
-                    pdh.addCol(row, "Patient_ID", processString(personDemos.getIdentifiers()).get("e1731641-30ab-102d-86b0-7a5022ba4115"));
+                    List<Object[]> drugOrder = results.stream().filter(Object -> Object[0].equals(i)).collect(Collectors.toList());
+                    for (Object[] o : drugOrder) {
+                        List<PersonDemographics> personDemographics = demographics.get(o[1]);
+                        PersonDemographics personDemos = personDemographics != null && personDemographics.size() > 0 ? personDemographics.get(0) : new PersonDemographics();
 
+                        int question = (int) o[2];
+                        String drug = String.valueOf(o[4]);
```
Comment (solemabrothers) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/DispensingDatasetDefinitionEvaluator.java]: @dbaluku the statement `String drug = String.valueOf(o[4]);` looks like it is not used any where, please remove it, if its not used

File: omod/src/main/resources/apps/ugandaemr_reports_app.json (hunk 2 above)
Comment (dbaluku) [reply omod/src/main/resources/apps/ugandaemr_reports_app.json]: @solemabrothers there is no dispensinglof report role created ye, that is a place holder

File: omod/src/main/resources/apps/ugandaemr_reports_app.json (hunk 1 above)
Comment (dbaluku) [reply omod/src/main/resources/apps/ugandaemr_reports_app.json]: Done

File: api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/DispensingDatasetDefinitionEvaluator.java (hunk 3 above)
Comment (dbaluku) [reply api/src/main/java/org/openmrs/module/ugandaemrreports/definition/dataset/evaluator/DispensingDatasetDefinitionEvaluator.java]: Done

**Comment by ssmusoke** [comment]
@slubwama @solemabrothers is there any facility that has been using the POC dispensing feature that we can use to validate the output of this report

**Pull request merged**